jcl/src/java.base/share/classes/java/lang/String.java

```diff
@@ -3874,16 +3878,60 @@ static void checkBoundsBeginEnd(int begin, int end, int length) {
 		throw newStringIndexOutOfBoundsException(begin, end, length);
 	}
 	
+	class StringSpliteratorOfInt implements Spliterator.OfInt {
+		int start; // index of first character in the array
+		int end; // index after the final character
+		int cursor; // next character to read
+		IntFunction func;
```

Comment:
This class should be `private static final` and all its fields should be `private final`.
Using `IntUnaryOperator` will avoid the boxing/unboxing implied via `IntFunction`.